forest/cli/src/cli/snapshot_cmd.rs

```diff
@@ -323,6 +348,123 @@ fn clean(config: &Config, snapshot_dir: &Option<PathBuf>, force: bool) -> anyhow
     Ok(())
 }
 
+async fn validate(
+    config: &Config,
+    validate_height: &i64,
+    snapshot: &PathBuf,
+) -> anyhow::Result<()> {
+    // 0. TODO: Prompt user of additional disk space usage.
+    let confirm = Confirm::with_theme(&ColorfulTheme::default())
+        .with_prompt(format!(
+            "This will result in using approximately {} MB of data. Proceed?",
+            std::fs::metadata(snapshot).unwrap().len() / (1024 * 1024)
+        ))
+        .default(false)
+        .interact()
+        .unwrap_or_default();
+
+    if confirm {
+        // clear temp db if exists
+        let tmp_db_path = PathBuf::from("/tmp/validate_snapshot");
+        let _ = std::fs::remove_dir_all(&tmp_db_path);
+
+        let db_path = tmp_db_path.join(&config.chain.name);
+        // 1. load car into a temporary database
+        let db = forest_db::rocks::RocksDb::open(db_path, &config.rocks_db)
+            .expect("Opening RocksDB must succeed");
+
+        let chain_store = Arc::new(ChainStore::new(db).await);
+
+        let genesis = read_genesis_header(
+            config.client.genesis_file.as_ref(),
+            config.chain.genesis_bytes(),
+            &chain_store,
+        )
+        .await
+        .unwrap();
+        let genesis_header = &genesis.blocks()[0];
+        chain_store.set_genesis(genesis_header).unwrap();
+
+        let sm = StateManager::new(
+            Arc::clone(&chain_store),
+            Arc::clone(&config.chain),
+            Arc::new(forest_interpreter::RewardActorMessageCalc),
+        )
+        .await
+        .unwrap();
+
+        let state_manager = Arc::new(sm);
+
+        let cids = {
+            let file = File::open(&snapshot).await?;
+            let reader = FetchProgress::fetch_from_file(file).await?;
+            load_and_retrieve_header(state_manager.blockstore(), reader, false).await?
```

Comment:
It's probably better just to call `load_car` here.

Reply:
Done